components/dashboard/src/usage/download/DownloadUsage.tsx

```diff
@@ -0,0 +1,135 @@
+import { Button } from "../../components/Button";
+import { useDownloadUsageCSV } from "./download-usage-csv";
+import { AttributionId } from "@gitpod/gitpod-protocol/lib/attribution";
+import { Dayjs } from "dayjs";
+import { useToast } from "../../components/toasts/Toasts";
+import { useCurrentOrg } from "../../data/organizations/orgs-query";
+import { ReactComponent as DownloadIcon } from "../../icons/Download.svg";
+import StatusDoneIcon from "../../icons/StatusDone.svg";
+import { ReactComponent as ExclamationIcon } from "../../images/exclamation.svg";
+import { LinkButton } from "../../components/LinkButton";
+import { saveAs } from "file-saver";
+import prettyBytes from "pretty-bytes";
+import { ProgressBar } from "../../components/ProgressBar";
+
+type Props = {
+    attributionId: AttributionId;
+    startDate: Dayjs;
+    endDate: Dayjs;
+};
+export const DownloadUsage: FC<Props> = ({ attributionId, startDate, endDate }) => {
+    const { data: org } = useCurrentOrg();
+    const { toast } = useToast();
+
+    const handleDownload = useCallback(async () => {
+        if (!org) {
+            return;
+        }
+
+        toast(
+            <DownloadUsageToast
+                orgName={org?.slug ?? org?.id}
+                attributionId={attributionId}
+                startDate={startDate}
+                endDate={endDate}
+            />,
+            {
+                autoHide: false,
+            },
+        );
+    }, [attributionId, endDate, org, startDate, toast]);
+
+    return (
+        <Button type="secondary" onClick={handleDownload} className="flex flex-row" icon={<DownloadIcon />}>
+            Export as CSV
+        </Button>
+    );
+};
+
+type DownloadUsageToastProps = Props & {
+    orgName: string;
+};
+
+const DownloadUsageToast: FC<DownloadUsageToastProps> = ({ attributionId, endDate, startDate, orgName }) => {
+    const [progress, setProgress] = useState(0);
+
+    const queryArgs = useMemo(
+        () => ({
+            orgName,
+            attributionId: AttributionId.render(attributionId),
+            from: startDate.startOf("day").valueOf(),
+            to: endDate.endOf("day").valueOf(),
+            onProgress: setProgress,
+        }),
+        [attributionId, endDate, orgName, startDate],
+    );
+    const { data, error, isLoading, abort, remove } = useDownloadUsageCSV(queryArgs);
+
+    const saveFile = useCallback(() => {
+        if (!data || !data.blob) {
+            return;
+        }
+
+        saveAs(data.blob, data.filename);
+    }, [data]);
+
+    useEffect(() => {
+        return () => {
+            abort();
+            remove();
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    if (isLoading) {
+        return (
+            <div className="space-y-2">
+                <div className="flex flex-row items-center space-x-2">
+                    <span>Preparing usage export</span>
```

Comment:
**question**: This is showing also when there are no reports for the selected date range. Could we skip this?

Reply:
Possibly, let me take a look. We don't know that there aren't any records until we've started downloading. Maybe I can wait until we have initial progress before showing that, because at that time we'd know if there are or aren't records.